### PR TITLE
Update the even-oddness of random numbers in getRandBitShare

### PR DESCRIPTION
**Pull request by mdonaka**
# Summary
Update the even-oddness of random numbers in getRandBitShare

# Purpose
Prevent re-generating random numbers

# Contents
- Update to generate odd numbers for SP and even numbers for NP

# Testing Methods Performed
- CI
- make test t=ComputationContainer p=medium
- Measure the number of regenerations

10000 horizontal joins．1 trial.

|version|#regen|
|-|-|
|previous versions(ae29a9ed8105f44c7bca8c24b056609cdd9d1137)|390|
|current version(c745c904f57a1f7befe7f0c5fe68ea0213c18226)|39871|
|this PR version|**0**|

### CODE REVIEW

File: src/ComputationContainer/Share/Share.hpp [hunk 1]
```diff
@@ -470,9 +470,18 @@ auto _sqrt(const T &x)
...
     while (true)
     {
-        auto r = Share<BT>(RandGenerator::getInstance()->getRand<BT>(-10, 10));
+        auto r_int = RandGenerator::getInstance()->getRand<int>(-10, 10);
+        // SPは奇数，それ以外は偶数になるまで再生成
+        while ((conf->party_id == conf->sp_id && !(r_int & 1))
+               || (conf->party_id != conf->sp_id && (r_int & 1)))
+        {
+            r_int = RandGenerator::getInstance()->getRand<int>(-10, 10);
+        }
```
Comment (arukuka) [inline src/ComputationContainer/Share/Share.hpp:482]: If we take the method of creating odd and even numbers by arithmetic, we don't need to regenerate random numbers.

```suggestion
        // this suggestion is not tested
        auto r_int = RandGenerator::getInstance()->getRand<int>(-5, 5) * 2;  // generate even number
        if (conf->party_id == conf->sp_id)
        {
            ++r_int;  // note: this range number range is `[-9, 11]`
        }
```

On the other hand, the cost is small compared to others, so I think it's okay to leave it as it is.
@mdonaka 
What do you think?

**Review by arukuka — COMMENTED** [review]
> version | #regen
> -- | --
> previous versions(ae29a9e) | 390
> current version(c745c90) | 39871
> this PR version | 0

Great!!  thank you for your contribution.  
I commented on the part that I was curious about, so please check it.

File: src/ComputationContainer/Share/Share.hpp (hunk 1 above)
Comment (mdonaka) [reply src/ComputationContainer/Share/Share.hpp:482]: Thanks for the suggestion! This one is much more beautiful!
No cost to fix, so I'll see if it works and fix it.

Comment (mdonaka) [reply src/ComputationContainer/Share/Share.hpp]: @arukuka 
Update with 31bd3302a519dd5e5b50fa9c4db8c5d354523904! Could you check it?

Comment (arukuka) [reply src/ComputationContainer/Share/Share.hpp]: @mdonaka 
Cool!  Thank you!

**Review by arukuka — APPROVED** [review]

**Pull request merged**